_distutils_hack/__init__.py

```diff
@@ -8,6 +8,11 @@
 is_pypy = '__pypy__' in sys.builtin_module_names
 
 
+warnings.filterwarnings('ignore',
+                        '.+ distutils .+ deprecated',
+                        DeprecationWarning)
```

Comment:
Can you describe what situation this resolves in the commit message or comment (or link to a description that summarizes the failure mode it addresses)? I'm unable to test on Python 3.10 due to other issues. I'm uneasy adding this change here because it will essentially suppress distutils deprecations anytime Setuptools is imported... and that could mask deprecations that need to be present.

Reply:
My understanding is that setuptools wants to carry their own distutils copy. This switch will happen before distutils is removed from Python, so the deprecation of distutils is well known to setuptools and the users of setuptools don't need to worry about it. Obviously, when users import both distutils and setuptools, they should be warned, but I am not sure how to handle that properly. As for the problems this resolves:

- https://bugzilla.redhat.com/show_bug.cgi?id=1927192
- https://bugzilla.redhat.com/show_bug.cgi?id=1927173

```

==================================== ERRORS ====================================
____________ ERROR collecting tests/unit/test_setuptools_command.py ____________
tests/unit/test_setuptools_command.py:3: in <module>
    from setuptools import dist
/usr/lib/python3.10/site-packages/setuptools/__init__.py:10: in <module>
    import distutils.core
/usr/lib64/python3.10/distutils/__init__.py:16: in <module>
    warnings.warn("The distutils package deprecated and slated for "
E   DeprecationWarning: The distutils package deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.48s ===============================
```

```
==================================== ERRORS ====================================
______________ ERROR collecting test/test_dependency_metadata.py _______________
test/test_dependency_metadata.py:6: in <module>
    from colcon_ros.package_identification.ros import _create_metadata
colcon_ros/package_identification/ros.py:16: in <module>
    from colcon_python_setup_py.package_identification.python_setup_py \
/usr/lib/python3.10/site-packages/colcon_python_setup_py/package_identification/python_setup_py.py:9: in <module>
    import setuptools
/usr/lib/python3.10/site-packages/setuptools/__init__.py:10: in <module>
    import distutils.core
/usr/lib64/python3.10/distutils/__init__.py:16: in <module>
    warnings.warn("The distutils package deprecated and slated for "
E   DeprecationWarning: The distutils package deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
=========================== short test summary info ============================
ERROR test/test_dependency_metadata.py - DeprecationWarning: The distutils pa...
```

Reply:
The tests on Setuptools still pass without the commit, so it seems that it's only through other packages that these warnings are emitted. More importantly, these errors are expected and the recommended workaround long-term is to use `SETUPTOOLS_USE_DISTUTILS=local` (which we know doesn't work on Linux yet). So maybe the best bet would be for those environments to suppress the warning rather than asking Setuptools to do so. If Setuptools suppresses the warning now, things may break later. I guess that's acceptable, given that the distutils platform problems are likely to be resolved by then.

Reply:
I still don't understand why Setuptools tests don't encounter this failure mode.